test/vanilla/ignite-new.test.ts

```diff
@@ -1,70 +1,349 @@
+      }
+
+      verifySplashScreenColor("android", `#191015`)
+      verifySplashScreenColor("expo", `#191015`)
+      verifySplashScreenColor(
+        "ios",
+        `red="0.0980392156862745" green="0.0627450980392157" blue="0.0823529411764706"`,
+      )
+
+      const splashScreenGen = await runIgnite(
+        `generate splash-screen 000000  --skip-source-equality-validation`,
+        runOpts,
+      )
+
+      expect(splashScreenGen).toContain(`Generating Expo splash screens`)
+
+      if (androidProjectExists) {
+        expect(splashScreenGen).toContain(`Generating Android splash screen...`)
+      } else {
+        expect(splashScreenGen).toContain(`No output directory found for "Android"`)
+      }
+
+      if (iosProjectExists) {
+        expect(splashScreenGen).toContain(`Generating iOS splash screen...`)
+      } else {
+        expect(splashScreenGen).toContain(`No output directory found for "iOS"`)
+      }
+
+      splashScreenAssets.forEach((i) => {
+        expect(filesystem.exists(i) === "file").toBe(true)
+      })
+
+      verifySplashScreenColor("android", `#000000`)
+      verifySplashScreenColor("expo", `#000000`)
+      verifySplashScreenColor(
+        "ios",
+        `red="0.00000000000000" green="0.00000000000000" blue="0.00000000000000"`,
+      )
+
+      const inputFile = filesystem.path(appPath, "ignite/templates/splash-screen/logo.png")
+      expect(filesystem.exists(inputFile) === "file").toBe(true)
+      filesystem.remove(inputFile)
+      expect(filesystem.exists(inputFile) === "file").toBe(false)
+      await runIgnite(`generate --update`, runOpts)
+      expect(filesystem.exists(inputFile) === "file").toBe(true)
+      // #endregion
+
+      // #region Assert Changes Can Be Commit To Git
+      // commit the change
+      await run(`git add ./app/models ./app/components ./app.json ./assets/images`, runOpts)
+      if (iosProjectExists) {
+        await run(`git add ./ios/Foo/Images.xcassets/AppIcon.appiconset`, runOpts)
+        await run(`git add ./ios/Foo/Images.xcassets/BootSplashLogo.imageset`, runOpts)
+        await run(`git add ./ios/Foo/BootSplash.storyboard`, runOpts)
+      }
+      if (androidProjectExists) {
+        await run(`git add ./android/app/src/main/res`, runOpts)
+      }
+      await run(`git commit -m "generated test components & assets"`, runOpts)
+      // #endregion
+
+      // #region Assert package.json Scripts Can Be Run
+      // run the tests; if they fail, run will raise and this test will fail
+      await run(`npm run test`, runOpts)
+      await run(`npm run lint`, runOpts)
+      await run(`npm run compile`, runOpts)
+      expect(await run("git diff HEAD", runOpts)).toEqual("")
+      // #endregion
+
+      // we're done!
+    })
   })
 })
 
-const ignoreFolders = ["/xcuserdata", ".git", "node_modules", "Pods", "/build"]
-
 async function checkForLeftoverHelloWorld(filePath: string) {
+  const ignoreFolders = ["/xcuserdata", ".git", "node_modules", "Pods", "/build"]
```

Comment:
We don't share this anywhere else, I made it local to the function so it will fold in VSCode